libraries/standard/http/src/private/http_client_parse.h

```diff
@@ -0,0 +1,72 @@
+#ifndef HTTP_CLIENT_PARSE_H_
+#define HTTP_CLIENT_PARSE_H_
+
+#include "http_client.h"
+/* #include "http_parser.h" */
+
+/**
+ * @brief The state of the response message parsed after
+ * #HTTPClient_ParseResponse returns.
+ */
+typedef enum HTTPParsingState_t
+{
+    HTTP_PARSING_NONE = 0,   /**< The parser has not started reading any response. */
+    HTTP_PARSING_INCOMPLETE, /**< The parse found a partial reponse. */
+    HTTP_PARSING_COMPLETE    /**< The parser found the entire response. */
+} HTTPParsingState_t;
+
+/**
+ * The HTTP response parsing context.
+ */
+typedef struct HTTPParsingContext
+{
+    /* http-parser dependencies will be added in the next incremental change. */
+    /* Below will be un-commented when parsing is implemented. */
+    #if 0
+        http_parser httpParser; /**< Third-party http-parser context. */
+    #endif
+    HTTPParsingState_t state;   /**< The current state of the HTTP response parsed. */
+    HTTPClient_HeaderParsingCallback_t * pHeaderParsingCallback; /**< Callback to invoke with each header found. */
+} HTTPParsingContext_t;
+
+/**
+ * Initialize the HTTP Client response parsing context.
+ *
+ * @param[in] pParsingContext The state of response parsing to initialize.
+ * @param[in] pHeaderParsingCallback Callback to invoked for each complete
```

Comment:
(Nit) Suggest "**The** callback **that will be** invoked for each header and value **pair** found **in parsing**."